### ISSUE TITLE
Nextcloud Website: Internal server error

### ISSUE THREAD

**Issue by ernoj** [nextcloud-snap/nextcloud-snap#2986]
#### Describe the bug
When I call up the Nextcloud website, I get to the login window. After logging in, however, I get the error message Internal server error.
The only thing I knowingly did before is
Update all apps with the command: sudo /snap/bin/nextcloud.occ app:update --all 
Update the php memory with the command:
sudo snap set nextcloud php.memory-limit=2048M
and then back to
sudo snap set nextcloud php.memory-limit=512M

#### Expected behavior
View my content.

#### OS/snapd/snap version
What OS and version are you running? Please also include the output of the
    installed: true
        version: 30.0.2.2
        versionstring: 30.0.2
        edition:
        maintenance: false
        needsDbUpgrade: false
        productname: Nextcloud
        extendedSupport: false
following commands:

    $ snap list nextcloud
Name       Version      Rev    Tracking       Publisher   Notes
nextcloud  30.0.2snap2  45417  latest/stable  nextcloud✓  -

and:

    $ snap version
snap    2.66.1+24.04
snapd   2.66.1+24.04
series  16
ubuntu  24.04
kernel  6.8.0-49-generic

#### Logs
See attachment
[log.txt](https://github.com/user-attachments/files/17978279/log.txt)

![Bildschirmfoto vom 2024-12-02 14-03-09](https://github.com/user-attachments/assets/3e191ebe-6dc0-4c30-83c7-38a149041a60)

**Comment by scubamuc** [comment]
Servus @ernoj and welcome

after scanning your logs, the only thing that really strikes me, is that you're blocking "NL" which you need for lets encryp to prevent this Lets Encrypt error:

```
[Mon Dec 02 10:01:33.271995 2024] [ssl:warn] [pid 3334:tid 3334] AH01909: 127.0.1.1:443:0 server certificate does NOT include an ID which matches the server name
AH00558: httpd: Could not reliably determine the server's fully qualified domain name, using 127.0.1.1. Set the 'ServerName' directive globally to suppress this message
[Mon Dec 02 10:01:33.274246 2024] [ssl:warn] [pid 3334:tid 3334] AH01909: 127.0.1.1:443:0 server certificate does NOT include an ID which matches the server name
[Mon Dec 02 10:01:33.274457 2024] [lbmethod_heartbeat:notice] [pid 3334:tid 3334] AH02282: No slotmem from mod_heartmonitor
```
see here https://github.com/nextcloud-snap/nextcloud-snap/wiki/Managing-HTTP-encryption-(HTTPS)#troubleshooting-multi-perspective-validation--geoblocking

----

since the error appeared after 
>Update all apps with the command: `sudo /snap/bin/nextcloud.occ app:update --all`

I'm guessing ``memories` could be the culprit regarding you apps

your logs:

```
        "memories": {
            "enabled": "yes",
            "installed_version": "7.4.1",
            "last_index_job_duration": "11.93",
            "last_index_job_start": "1733133190",
            "last_index_job_status": "Indexing completed successfully",
            "last_index_job_status_type": "success",
            "types": ""
        },
```

its hard to pinpoint you issue... 

* first try getting lets encrypt working properly by unblocking "NL" for correct [Lets Encrypt Multi-Perspective Validation & Geoblocking](https://community.letsencrypt.org/t/multi-perspective-validation-geoblocking-faq/218158)
* the try disabling `memories` see [FAQ's](https://github.com/nextcloud-snap/nextcloud-snap/wiki/FAQ's#memories-app)

**Comment by scubamuc** [comment]
@ernoj can you reach your Nextcloud snap locally without https?

**Comment by scubamuc** [comment]
hmmmm...

> Update the php memory with the command:
sudo snap set nextcloud php.memory-limit=2048M
and then back to
sudo snap set nextcloud php.memory-limit=512M

that shouldn't be an issue, there are no logs pointing to that

**Comment by ernoj** [comment]
Thanks for the quick analysis! I have a rule in the firewall to allow the IP address range for Lets Encrypt. Obviously it is not covered by this. I will check this again. 


I have now restored the Nextcloud from a backup and it is currently running again. However, I'm worried that the problem will occur again! Very strange that I can't find anything about this in the logs.


With the Memories app, I have the problem that Recognize is not running and the error "The Node.js library could not be executed. You may need to manually specify the path to a working library.". However, I have specified the path to Node.js. 

But that's probably not the problem, is it?

**Comment by scubamuc** [comment]
@ernoj 

> With the Memories app, I have the problem that Recognize is not running and the error "The Node.js library could not be executed. You may need to manually specify the path to a working library.". However, I have specified the path to Node.js.

i'm afraid this will be the issue since redefining the node.js path will fail regardless... 

   > [!IMPORTANT]
   > Latest Recognize versions, when in WASM mode, require Node.js 20, which in turn requires a version of glibc >= 2.28 (more context [here](https://github.com/nextcloud/recognize/issues/1130)). The nextcloud snap is still using glibc 2.27 (from core18), so Recognize's WASM mode won't work with the snap till [this issue](https://github.com/nextcloud-snap/nextcloud-snap/issues/2908) is solved.

so you will bump into this issue until  [this issue](https://github.com/nextcloud-snap/nextcloud-snap/issues/2908) is solved.

**Comment by scubamuc** [comment]
@ernoj for me personally?

 see https://scubamuc.github.io/

- i've been using the Nextcloud snap without issues since 2019 (first on Raspi3b) and [current setup](https://github.com/scubamuc/scubamuc.github.io) since 2020 non stop 24/7
- the Nextcloud snap community is a great place to hang out with plenty to learn from a great team (i also hang out with the [Nextcloud support](https://help.nextcloud.com/tag/snap) guys though)
- i manage 3 (private) instances and one corporate (128 users, LDAP) instance with the snap and never had to hassle with updates and upgrades
- i'm an Ubuntu user and snaps simply work with no issues

> Some people are wondering when they should prefer the snap over the VM or even Dockers which are available from other repositories.
> 
> Here is our answer:
> 
> The snap is nice for getting up and running quickly with minimal space, and will work great if you're happy not messing with it. It's a quality, optimized LAMP stack that will scale just as well as a "normal" Nextcloud installation; it's made for production. Since it's a snap you also get the benefit of automatic updates and the ability to rollback without worrying about needing to take a snapshot, etc.
> 
> However, the snap is very opinionated. Don't want to use Apache? Sorry, the snap uses it. Don't want to use MySQL? Sorry, that's what the snap uses. Something other than PHP whatever-we're-using? We picked the version we feel gives the best results. In other words, it's not very tinker-friendly. We don't do this to be mean, we do this so that we can reliably update it without your needing to worry about it.
> 
> The VM is much more flexible. It's a full version of Ubuntu server edition, allowing you to tweak whatever you need and it comes with many apps which are not that easy to configure for inexperienced administrators. This of course makes it larger. You'll also need to make sure you maintain it and keep the OS up-to-date. Since it's virtualized you can assign disk, CPU, memory, and network quotas to it (you'd need to install the snap in an lxc container or a VM to get the same abilities).

**Comment by scubamuc** [comment]
@ernoj 
> With regard to the recognize app, I'm obviously already reaching the limits of the Snap installation.

that depends on what's nice to have and what you really need. a default basic Nextcloud would fulfil 98% needs... the rest is nice to have. AI and photo management in the Nextcloud aren't priority.  those things can be managed locally on the NAS and [CODE & Nextcloud Office configuration](https://github.com/nextcloud-snap/nextcloud-snap/wiki/Configure-CODE-and-Nextcloud-office-for-Nextcloud-snap) is simple enough.

**Comment by ernoj** [comment]
A nice summary! Thank you.
The photo AI solution is really just nice to have. But it would be a nice addition and it worked in the beginning. Until it suddenly stopped working, but they're obviously working on that. Then I'll just wait and see.

As described above, I restored the Nextcloud from a backup.  The user data was stored on a NAS. The data is therefore still available. However, the fact that a backup was restored was problematic because the appointments and tasks were no longer up to date, but from the backup. Is it generally possible to have a backup of this as well, so that in the event of a Nextcloud crash (as happened to me) the software itself can be restored from a backup, but the user content can be kept completely up to date?

**Comment by scubamuc** [comment]
@ernoj 

>  so that in the event of a Nextcloud crash (as happened to me) the software itself can be restored from a backup, but the user content can be kept completely up to date?

yeah see [Backup, restore & migrate](https://github.com/nextcloud-snap/nextcloud-snap/wiki/Backup-and-Restore) as well as [Export & import](https://github.com/nextcloud-snap/nextcloud-snap/wiki/Export-and-Import)

> was problematic because the appointments and tasks were no longer up to date

that's not as simple since your contacts and calendar data is not a file per se but rather a database table which would need to be exported separately. with daily LXC snapshots you're not going to lose much data and its easy to spin up a clone and get your data from there... but if you're running the snap on the host without containers you'd probably be wise to keep a regular [database export](https://github.com/nextcloud-snap/nextcloud-snap/wiki/Export-and-Import#export-nextcloud-snap-with-nextcloudexport) using `sudo nextcloud.export -d` ... should work well using a [rotating Nextcloud snap export script](https://github.com/scubamuc/wiki-md/blob/scubamuc-wiki/NEXTCLOUD.snap--backup_nextcloud-export.md) in cron.

or

if you're syncing to android you can set daily backups using the Davx5 client... find this in the Nextcloud app for Android under settings (Nextcloud-->Einstellungen-->Mehr --> Kontakte- und Kalender Sicherung) thus copying a single addressbook file and calendar files to the (hidden) `.Calendar`and `.Contacts` directories:

![grafik](https://github.com/user-attachments/assets/598f6a76-54e0-4a9b-a8a1-d068e3815503)

**Comment by scubamuc** [comment]
@ernoj 

> export the database via “sudo nextcloud.export -d”.

nope... rather `sudo nextcloud.export -b`

please reread, i've edited since...

> du kannst in meinem GitHub ein issue in Deutsch schreiben... oder du kannst im [Nextcloud Help Forum](https://help.nextcloud.com/c/international/18) in Deutsch fragen

**Comment by ernoj** [comment]
<3

**Comment by scubamuc** [comment]
@ernoj

>  Therefore, as an admin, I am looking for a solution that works without the other members having to do anything.

maybe this works for you: [calcardbackup](https://codeberg.org/BernieO/calcardbackup)

**Comment by yannicklescure** [comment]
What about installing node.js in the ubuntu server and in nextcloud config.php:

```php
<?php
$CONFIG = array (
  'nodejs_binary' => '/usr/bin/node',  // or your custom path
);
```

otherwise https://github.com/vercel/pkg might help or even https://flox.dev/blog/using-flox-to-create-portable-reproducible-nodejs-environments/
